File: CPP_programming/Day8_Inheritance-2/A1_Wage_Manager_Salesperson/Employee.cpp
#include"Employee.h"


int Employee:: cnt=0;
Employee::Employee()///////////////////////constructor
{
    cnt++;
    e_id=cnt;
    sal=0;
}
Employee::Employee(const char* nm, double sal, int age):Person(nm,age)
{
    cnt++;
    e_id=cnt;
    this->sal=sal;
}
Employee::Employee(Employee &obj):Person(obj)///////////////copy constructor
{
    cnt++;
    this->e_id=cnt;
    this->sal=obj.sal;
    //strcpy(this->name,obj.name);//in this line there is no object of person created directly so we cant call person's variables with obj.name
    //this->bdate=obj.bdate
}
 
void Employee::display()//in this function display() is overidden
{
    cout<<"Employee Id: "<<e_id<<"\n";
    Person::display();
    cout<<"Employee Base Salary: "<<sal<<"\n\n";
}

File: Core_java_new/_09Interface/A3_Employee_usingInterface/employee/Employee.java
package employee;

public class Employee extends Person {
	//private Person p1;
			public int eid;
			protected double salary;
			
			public Employee() {
				super();
				eid=1;
				salary=0;
			}
			
			public Employee(String name,int dd,int mm,int yy, int eid, double salary) {
				//super();
				//this.p1 = p1;
				super(name,dd,mm,yy);
				this.eid = eid;
				this.salary = salary;
			}
		
			public void display()
			{
				super.display();
				System.out.println("Emp ID: "+eid);
				
				System.out.println("Salary: "+salary);
				System.out.println();
				System.out.print("abc\n");
				
			}
}
